numpy/core/shape_base.py

```diff
@@ -187,22 +178,106 @@ def atleast_3d(*arys):
     [[[1 2]]] (1, 1, 2)
 
     """
-    res = []
-    for ary in arys:
-        ary = asanyarray(ary)
-        if ary.ndim == 0:
-            result = ary.reshape(1, 1, 1)
-        elif ary.ndim == 1:
-            result = ary[_nx.newaxis, :, _nx.newaxis]
-        elif ary.ndim == 2:
-            result = ary[:, :, _nx.newaxis]
-        else:
-            result = ary
-        res.append(result)
-    if len(res) == 1:
-        return res[0]
-    else:
-        return res
+    return _atleast_nd_result([atleast_nd(atleast_nd(arys, 2, 0), 3, -1)
+                               for a in arys])
+
+
+def _atleast_nd_dispatcher(ary, ndim, pos=None):
```

Comment:
I am not 100% sure how dispatchers work, but this *seems* to be right. Someone should double-check it though.

Reply:
Apparently it always has to be a sequence: `(ary,)`